cypress/e2e/swap/settings.test.ts

```diff
@@ -16,14 +16,24 @@ describe('Swap settings', () => {
   })
 
   it('should open the mobile settings menu', () => {
+    // Set viewport to iPhone 6
     cy.viewport('iphone-6')
+
+    // Visit the swap page with specific feature flag
     cy.visit('/swap', { featureFlags: [FeatureFlag.uniswapXEnabled] })
-    cy.get(getTestSelector('open-settings-dialog-button')).click()
-    cy.get(getTestSelector('mobile-settings-menu')).should('exist')
-    cy.contains('Max slippage').should('exist')
-    cy.contains('Transaction deadline').should('exist')
-    cy.contains('UniswapX').should('exist')
-    cy.contains('Local routing').should('exist')
-    cy.get(getTestSelector('mobile-settings-scrim')).click()
+
```

Comment:
can't you just do this `cy.get(getTestSelector('web3-status-connected'))` to wait for it to be connected?

Reply:
Yeah, but the underlying behavior that was causing this flake is actually an issue in our app. If you move quickly after page load you can hit this bug in prod today.

Reply:
This pr does resolve the bug. It also updates the test to be easier to read and more idiomatic.